src/libraries/Microsoft.Extensions.Configuration.FileExtensions/src/FileConfigurationProvider.cs

```diff
@@ -80,16 +82,35 @@ private void Load(bool reload)
                 {
                     Data = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
                 }
-                using (var stream = file.CreateReadStream())
+
+                static Stream OpenRead(IFileInfo fileInfo)
```

Comment:
Maybe this is a stylistic thing, but is OpenRead simple enough to just inline?

```
// The default physical file info assumes asynchronous IO which results in unnecessary overload
using Stream stream = (file.PhysicalPath != null) ? new FileStream(...) : file.CreateReadStream()
```

Reply:
too many arguments to the FileStream ctor. I had this at first.